Plugins/FastJet/include/Acts/Plugins/FastJet/TrackJets.hpp

```diff
@@ -0,0 +1,71 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2024 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include <Acts/Definitions/Units.hpp>
+
+#include <optional>
+#include <vector>
+
+#include <fastjet/JetDefinition.hh>
+#include <fastjet/PseudoJet.hh>
+
+namespace Acts::FastJet {
+
+template <typename TrackContainer>
+class TrackJetSequence {
+ public:
+  /// Get all the track jets passing the pT & eta cuts
+  ///
+  /// @param ptMin the minimum jet pT in GeV
+  /// @param etaMax the maximum jet absolute eta
+  ///
+  /// @return a vector of fastjet::PseudoJet objects
+  std::vector<fastjet::PseudoJet> jets(float ptMin = 20 *
+                                                     Acts::UnitConstants::GeV,
+                                       float etaMax = 2.5);
+
+  /// Get the tracks making up a track-jet
+  ///
+  /// @param jet the jet from which to get the constituent tracks
+  /// @param coreR optional radius inside which to get the tracks
+  ///
+  /// @return a vector of TrackProxy
+  std::vector<typename TrackContainer::TrackProxy> tracksInJet(
+      const fastjet::PseudoJet& jet, std::optional<float> coreR = {});
+
+  /// Main constructor, but using the "makeTrackJets" function is recommended
+  ///
+  /// @param clusterSeq the fastjet::ClusterSequence object
+  /// @param inputTracks the input tracks that make up the sequence
+  TrackJetSequence(fastjet::ClusterSequence clusterSeq,
+                   TrackContainer& inputTracks)
+      : m_clusterSeq{std::move(clusterSeq)}, m_inputTracks{inputTracks} {}
```

Comment:
Consider making the constructor private and the `makeTrackJets` function a static member, or alternatively a constructor itself. I think having a factory instead of a constructor is maybe more suitable, because it does run the (non-trivial) jet clustering algorithm when called, which might be a bit unexpected if it's just a constructor.

Reply:
Done